tests/xharness/Jenkins/Reports/MarkdownReportWriter.cs

```diff
@@ -16,19 +16,26 @@ void WriteFailedTestsDetails (IEnumerable<ITestTask> failedTests, StreamWriter w
 			writer.WriteLine ("## Failed tests");
 			writer.WriteLine ();
 
-			foreach (var test in failedTests) {
-				writer.Write ($" * {test.TestName}");
-				if (!string.IsNullOrEmpty (test.Mode))
-					writer.Write ($"/{test.Mode}");
-				if (!string.IsNullOrEmpty (test.Variation))
-					writer.Write ($"/{test.Variation}");
-				writer.Write ($": {test.ExecutionResult}");
-				if (!string.IsNullOrEmpty (test.FailureMessage))
-					writer.Write ($" ({test.FailureMessage})");
-				if (test.KnownFailure.HasValue)
-					writer.Write ($" Known issue: [{test.KnownFailure.Value.HumanMessage}]({test.KnownFailure.Value.IssueLink})");
-				writer.WriteLine ();
+			foreach (var group in failedTests.GroupBy ((v) => v.TestName)) {
+				var enumerableGroup = group as IEnumerable<ITestTask>;
+				if (enumerableGroup != null) {
```

Comment:
looks like a candidate for `if (group is IEnumerable<ITestTask> enumerableGroup) {`